# Download dataset

In [1]:
# Download dataset from Kaggle using kagglehub library
import kagglehub

path = kagglehub.dataset_download("piterfm/milano-cortina-2026-olympic-winter-games")

print("Path to dataset files:", path)

100%|██████████| 101k/101k [00:00<00:00, 524kB/s]

Extracting files...


Path to dataset files: /home/thomas/.cache/kagglehub/datasets/piterfm/milano-cortina-2026-olympic-winter-games/versions/11


# Initial dataset investigation

## Athletes

In [ ]:
# Load the dataset using pandas
import pandas as pd

df_athletes = pd.read_csv(path + "/athletes.csv")

In [9]:
df_athletes.shape

(2916, 6)

In [8]:
df_athletes.info()

<class 'pandas.DataFrame'>
RangeIndex: 2916 entries, 0 to 2915
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   code          2916 non-null   int64
 1   name          2916 non-null   str  
 2   gender        2916 non-null   str  
 3   country_code  2916 non-null   str  
 4   flag_bearer   2916 non-null   int64
 5   events        2916 non-null   str  
dtypes: int64(2), str(4)
memory usage: 136.8 KB


In [5]:
df_athletes.head()

,code,name,gender,country_code,flag_bearer,events
0,139893,AABREKK Ingrid Bergene,F,NOR,0,"[{'discipline': 'Cross-Country Skiing', 'event..."
1,24464,AAGAARD Mikkel,M,DEN,0,"[{'discipline': 'Ice Hockey', 'event': 'Men'}]"
2,45083,AALTO Antti,M,FIN,0,"[{'discipline': 'Ski Jumping', 'event': ""Men's..."
3,32319,ABATANGELO Aurora,F,ITA,0,"[{'discipline': 'Ice Hockey', 'event': 'Women'}]"
4,4972,ABDI Fayik,M,KSA,0,"[{'discipline': 'Alpine Skiing', 'event': ""Men..."


In [6]:
df_athletes.tail()

,code,name,gender,country_code,flag_bearer,events
2911,25116,ZUMWINKLE Grace,F,USA,0,"[{'discipline': 'Ice Hockey', 'event': 'Women'}]"
2912,44328,ZUNTE Darta,F,EST,0,"[{'discipline': 'Skeleton', 'event': 'Women'}]"
2913,7319,ZURBAY Anabelle,F,IRL,1,"[{'discipline': 'Alpine Skiing', 'event': ""Wom..."
2914,19124,ZURBRIGGEN Anina,F,BUL,0,"[{'discipline': 'Alpine Skiing', 'event': ""Wom..."
2915,23401,ZUREK Damian,M,POL,0,"[{'discipline': 'Speed Skating', 'event': ""Men..."


In [10]:
df_athletes.describe()

,code,flag_bearer
count,2916.000000,2916.000000
mean,36401.582990,0.122085
std,16043.661964,0.470224
min,4098.000000,0.000000
25%,24173.000000,0.000000
50%,39906.500000,0.000000
75%,45897.000000,0.000000
max,170798.000000,3.000000


## Coaches

In [ ]:
# Load the dataset using pandas
import pandas as pd

df_coaches = pd.read_csv(path + "/coaches.csv")

In [ ]:
df_coaches.shape

(50, 7)

In [ ]:
df_coaches.info()

<class 'pandas.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   code          50 non-null     int64
 1   name          50 non-null     str  
 2   gender        50 non-null     str  
 3   country_code  50 non-null     str  
 4   nationality   50 non-null     str  
 5   function      50 non-null     str  
 6   events        50 non-null     str  
dtypes: int64(1), str(6)
memory usage: 2.9 KB


In [ ]:
df_coaches.head()

,code,name,gender,country_code,nationality,function,events
0,50140,BOUCHARD Eric,M,ITA,CAN,Head Coach,"[{'discipline': 'Ice Hockey', 'event': 'Women'}]"
1,50988,CERNOVSKY Vladimir,M,CZE,CZE,Coach,"[{'discipline': 'Curling', 'event': 'Mixed Dou..."
2,49020,CHARETTE Pierre,M,SUI,CAN,Coach,"[{'discipline': 'Curling', 'event': 'Women'}]"
3,53809,COOPER Jon,M,CAN,CAN,Head Coach,"[{'discipline': 'Ice Hockey', 'event': 'Men'}]"
4,49035,de CRUZ Peter,M,SUI,SUI,Coach,"[{'discipline': 'Curling', 'event': 'Mixed Dou..."


In [ ]:
df_coaches.tail()

,code,name,gender,country_code,nationality,function,events
45,47166,VITOLINS Harijs,M,LAT,LAT,Head Coach,"[{'discipline': 'Ice Hockey', 'event': 'Men'}]"
46,22959,WALSTAD Steffen,M,EST,NOR,Coach,"[{'discipline': 'Curling', 'event': 'Mixed Dou..."
47,53327,WEBSTER Paul,M,CAN,CAN,Coach,"[{'discipline': 'Curling', 'event': 'Men'}]"
48,25178,WROBLEWSKI John,M,USA,USA,Head Coach,"[{'discipline': 'Ice Hockey', 'event': 'Women'}]"
49,39951,ZANG Jialiang,M,CHN,CHN,Coach,"[{'discipline': 'Curling', 'event': 'Women'}]"


In [ ]:
df_coaches.describe()

,code
count,50.000000
mean,41325.620000
std,19130.461623
min,18095.000000
25%,29014.000000
50%,41103.500000
75%,49326.250000
max,149123.000000


## Medallists

In [ ]:
# Load the dataset using pandas
import pandas as pd

df_medallists = pd.read_csv(path + "/medallists.csv")

In [ ]:
df_medallists.shape

(709, 11)

In [ ]:
df_medallists.info()

<class 'pandas.DataFrame'>
RangeIndex: 709 entries, 0 to 708
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   date             709 non-null    str  
 1   medal_code       709 non-null    int64
 2   medal            709 non-null    str  
 3   code             709 non-null    int64
 4   name             709 non-null    str  
 5   gender           709 non-null    str  
 6   country_code     709 non-null    str  
 7   country          709 non-null    str  
 8   discipline       709 non-null    str  
 9   discipline_code  709 non-null    str  
 10  event_name       709 non-null    str  
dtypes: int64(2), str(9)
memory usage: 61.1 KB


In [ ]:
df_medallists.head()

,date,medal_code,medal,code,name,gender,country_code,country,discipline,discipline_code,event_name
0,2026-02-07,1,GOLD,48729,von ALLMEN Franjo,M,SUI,Switzerland,Alpine Skiing,ALP,Men's Downhill
1,2026-02-07,2,SILVER,25998,FRANZONI Giovanni,M,ITA,Italy,Alpine Skiing,ALP,Men's Downhill
2,2026-02-07,3,BRONZE,26002,PARIS Dominik,M,ITA,Italy,Alpine Skiing,ALP,Men's Downhill
3,2026-02-07,1,GOLD,30739,KARLSSON Frida,F,SWE,Sweden,Cross-Country Skiing,CCS,Women's 10km + 10km Skiathlon
4,2026-02-07,2,SILVER,30707,ANDERSSON Ebba,F,SWE,Sweden,Cross-Country Skiing,CCS,Women's 10km + 10km Skiathlon


In [ ]:
df_medallists.tail()

,date,medal_code,medal,code,name,gender,country_code,country,discipline,discipline_code,event_name
704,2026-02-22,2,SILVER,53901,SUZUKI Nick,M,CAN,Canada,Ice Hockey,IHO,Men
705,2026-02-22,2,SILVER,53905,THEODORE Shea,M,CAN,Canada,Ice Hockey,IHO,Men
706,2026-02-22,2,SILVER,53907,THOMPSON Logan,M,CAN,Canada,Ice Hockey,IHO,Men
707,2026-02-22,2,SILVER,53910,TOEWS Devon,M,CAN,Canada,Ice Hockey,IHO,Men
708,2026-02-22,2,SILVER,53914,WILSON Tom,M,CAN,Canada,Ice Hockey,IHO,Men


In [ ]:
df_medallists.describe()

,medal_code,code
count,709.000000,709.000000
mean,1.994358,37871.267983
std,0.814745,14925.077758
min,1.000000,6464.000000
25%,1.000000,25029.000000
50%,2.000000,42254.000000
75%,3.000000,48300.000000
max,3.000000,170798.000000


## Medals

In [ ]:
# Load the dataset using pandas
import pandas as pd

df_medals = pd.read_csv(path + "/medals.csv")

In [ ]:
df_medals.shape

(29, 8)

In [ ]:
df_medals.info()

<class 'pandas.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   country       29 non-null     str  
 1   country_code  29 non-null     str  
 2   gold          29 non-null     int64
 3   silver        29 non-null     int64
 4   bronze        29 non-null     int64
 5   total         29 non-null     int64
 6   rank          29 non-null     int64
 7   rank_total    29 non-null     int64
dtypes: int64(6), str(2)
memory usage: 1.9 KB


In [ ]:
df_medals.head()

,country,country_code,gold,silver,bronze,total,rank,rank_total
0,Norway,NOR,18,12,11,41,1,1
1,United States,USA,12,12,9,33,2,2
2,Netherlands,NED,10,7,3,20,3,9
3,Italy,ITA,10,6,14,30,4,3
4,Germany,GER,8,10,8,26,5,4


In [ ]:
df_medals.tail()

,country,country_code,gold,silver,bronze,total,rank,rank_total
24,Denmark,DEN,0,1,0,1,25,24
25,Estonia,EST,0,1,0,1,25,24
26,Georgia,GEO,0,1,0,1,25,24
27,Bulgaria,BUL,0,0,2,2,28,22
28,Belgium,BEL,0,0,1,1,29,24


In [ ]:
df_medals.describe()

,gold,silver,bronze,total,rank,rank_total
count,29.000000,29.000000,29.000000,29.000000,29.000000,29.000000
mean,4.000000,4.034483,3.965517,12.000000,14.862069,14.241379
std,4.511889,3.877432,4.083790,11.655777,8.356880,7.744377
min,0.000000,0.000000,0.000000,1.000000,1.000000,1.000000
25%,0.000000,1.000000,1.000000,2.000000,8.000000,8.000000
50%,3.000000,2.000000,2.000000,6.000000,15.000000,14.000000
75%,6.000000,7.000000,6.000000,21.000000,22.000000,22.000000
max,18.000000,12.000000,14.000000,41.000000,29.000000,24.000000


## Schedules

In [ ]:
# Load the dataset using pandas
import pandas as pd

df_schedules = pd.read_csv(path + "/schedules.csv")

In [ ]:
df_schedules.shape

(1884, 16)

In [ ]:
df_schedules.info()

<class 'pandas.DataFrame'>
RangeIndex: 1884 entries, 0 to 1883
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   start_date       1884 non-null   str  
 1   end_date         1878 non-null   str  
 2   day              1884 non-null   str  
 3   status           1884 non-null   str  
 4   discipline       1884 non-null   str  
 5   discipline_code  1884 non-null   str  
 6   event            1884 non-null   str  
 7   event_medal      1884 non-null   int64
 8   phase            1884 non-null   str  
 9   gender           1884 non-null   str  
 10  event_type       1884 non-null   str  
 11  venue            1884 non-null   str  
 12  venue_code       1884 non-null   str  
 13  location         1884 non-null   str  
 14  location_code    1884 non-null   str  
 15  id               1884 non-null   str  
dtypes: int64(1), str(15)
memory usage: 235.6 KB


In [ ]:
df_schedules.head()

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
0,2026-02-04T11:30:00+01:00,2026-02-04T13:30:00+01:00,2026-02-04,FINISHED,Alpine Skiing,ALP,Men's Downhill,0,Men's Downhill Official Training,M,INDV,Stelvio Ski Centre,SSC,Stelvio Ski Centre-Alpine Skiing Course,SAL,ALPMDH----------------TRNO000100--
1,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet A,CUA,CURXTEAM2-------------PREL000101--
2,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet B,CUB,CURXTEAM2-------------PREL000102--
3,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet C,CUC,CURXTEAM2-------------PREL000103--
4,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet D,CUD,CURXTEAM2-------------PREL000104--


In [ ]:
df_schedules.tail()

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
1879,2026-02-22T11:09:00+01:00,2026-02-22T11:34:00+01:00,2026-02-22,FINISHED,Freestyle Skiing,FRS,Women's Freeski Halfpipe,0,Women's Freeski Halfpipe Final,W,INDV,Livigno Snow Park,LSP,Livigno Snow Park-Halfpipe,LHP,FRSWHP----------------FNL-000200--
1880,2026-02-22T11:38:00+01:00,2026-02-22T12:03:00+01:00,2026-02-22,FINISHED,Freestyle Skiing,FRS,Women's Freeski Halfpipe,1,Women's Freeski Halfpipe Final,W,INDV,Livigno Snow Park,LSP,Livigno Snow Park-Halfpipe,LHP,FRSWHP----------------FNL-000300--
1881,2026-02-22T12:15:00+01:00,2026-02-22T13:00:00+01:00,2026-02-22,FINISHED,Bobsleigh,BOB,4-man,1,4-man Competition,M,TEAM,Cortina Sliding Centre,CSC,Cortina Sliding Centre,CSC,BOBMTEAM4-------------FNL-000400--
1882,2026-02-22T14:10:00+01:00,2026-02-22T16:40:00+01:00,2026-02-22,FINISHED,Ice Hockey,IHO,Men,1,Men's Play-off Round,M,TEAM,Milano Santagiulia Ice Hockey Arena,MH1,Milano Santagiulia Ice Hockey Arena,MH1,IHOMTEAM6-------------FNL-000100--
1883,2026-02-22T20:30:00+01:00,2026-02-22T23:00:00+01:00,2026-02-22,SCHEDULED,Ceremonies,CER,Closing Ceremony,0,Closing Ceremony,G,NONE,Verona Olympic Arena,VOA,Verona Olympic Arena,VOA,CERGCLOSING-----------OTHR--------


In [ ]:
df_schedules.describe()

,event_medal
count,1884.000000
mean,0.237261
std,0.534991
min,0.000000
25%,0.000000
50%,0.000000
75%,0.000000
max,3.000000


## Teams

In [ ]:
# Load the dataset using pandas
import pandas as pd

df_teams = pd.read_csv(path + "/teams.csv")

In [ ]:
df_teams.shape

(550, 6)

In [ ]:
df_teams.info()

<class 'pandas.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   code          550 non-null    str  
 1   name          550 non-null    str  
 2   team_type     550 non-null    str  
 3   gender        550 non-null    str  
 4   country_code  550 non-null    str  
 5   events        550 non-null    str  
dtypes: str(6)
memory usage: 25.9 KB


In [ ]:
df_teams.head()

,code,name,team_type,gender,country_code,events
0,FSKXPAIRS---ARM01,AKOPOVA Karina / RAKHMANIN Nikita,CPLW,-,ARM,"[{'discipline': 'Figure Skating', 'event': 'Pa..."
1,BOBMTEAM2---GER03,AMMOUR Adam,CUSTOM,-,GER,"[{'discipline': 'Bobsleigh', 'event': '2-man'}]"
2,BOBMTEAM4---GER03,AMMOUR Adam,CUSTOM,-,GER,"[{'discipline': 'Bobsleigh', 'event': '4-man'}]"
3,BOBWTEAM2---ITA01,ANDREUTTI Giada,CUSTOM,-,ITA,"[{'discipline': 'Bobsleigh', 'event': '2-woman'}]"
4,SKNXRELAY2--LAT01,ANDZANE Marta / INDRIKSONS Emils,CPLW,-,LAT,"[{'discipline': 'Skeleton', 'event': 'Mixed Te..."


In [ ]:
df_teams.tail()

,code,name,team_type,gender,country_code,events
545,BOBMTEAM2---NED01,WESSELINK Dave,CUSTOM,-,NED,"[{'discipline': 'Bobsleigh', 'event': '2-man'}]"
546,BOBWTEAM2---CHN02,YING Qing,CUSTOM,-,CHN,"[{'discipline': 'Bobsleigh', 'event': '2-woman'}]"
547,FSKXICEDANCEJPN01,YOSHIDA Utana / MORITA Masaya,CPLW,X,JPN,"[{'discipline': 'Figure Skating', 'event': 'Te..."
548,SKNXRELAY2--CHN02,ZHAO Dan / CHEN Wenhao,CPLW,-,CHN,"[{'discipline': 'Skeleton', 'event': 'Mixed Te..."
549,FSKXICEDANCEUSA02,ZINGAS Emilea / KOLESNIK Vadym,CPLW,-,USA,"[{'discipline': 'Figure Skating', 'event': 'Ic..."


In [ ]:
df_teams.describe()

,code,name,team_type,gender,country_code,events
count,550,550,550,550,550,550
unique,550,205,5,4,43,39
top,FSKXPAIRS---ARM01,United States of America,ORG,-,USA,"[{'discipline': 'Alpine Skiing', 'event': ""Wom..."
freq,1,24,329,242,46,28


Name is either individuals or country.

## Venues

In [133]:
# Load the dataset using pandas
import pandas as pd

df_venues = pd.read_csv(path + "/venues.csv")

In [134]:
df_venues.shape

(15, 6)

In [135]:
df_venues.info()

<class 'pandas.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   venue             15 non-null     str  
 1   venue_code        15 non-null     str  
 2   disciplines       13 non-null     str  
 3   disciplines_code  13 non-null     str  
 4   url_sport         7 non-null      str  
 5   url_venue         15 non-null     str  
dtypes: str(6)
memory usage: 852.0 bytes


In [136]:
df_venues.head()

,venue,venue_code,disciplines,disciplines_code,url_sport,url_venue
0,Anterselva Biathlon Arena,ABA,discipline-biathlon,BTH,https://www.olympics.com/en/sports/biathlon/,https://www.olympics.com/en/milano-cortina-202...
1,Cortina Curling Olympic Stadium,CCU,discipline-curling,CUR,https://www.olympics.com/en/sports/curling/,https://www.olympics.com/en/milano-cortina-202...
2,Cortina Sliding Centre,CSC,"[""discipline-bobsleigh"",""discipline-luge"",""dis...","[""BOB"",""LUG"",""SKN""]",NaN,https://www.olympics.com/en/milano-cortina-202...
3,Livigno Aerials & Moguls Park,LAM,discipline-freestyle-skiing,FRS,https://www.olympics.com/en/sports/freestyle-s...,https://www.olympics.com/en/milano-cortina-202...
4,Livigno Snow Park,LSP,"[""discipline-freestyle-skiing"",""discipline-sno...","[""FRS"",""SBD""]",NaN,https://www.olympics.com/en/milano-cortina-202...


In [137]:
df_venues.tail()

,venue,venue_code,disciplines,disciplines_code,url_sport,url_venue
10,Predazzo Ski Jumping Stadium,PSJ,"[""discipline-nordic-combined"",""discipline-ski-...","[""NCB"",""SJP""]",NaN,https://www.olympics.com/en/milano-cortina-202...
11,Stelvio Ski Centre,SSC,"[""discipline-alpine-skiing"",""discipline-ski-mo...","[""ALP"",""SMT""]",NaN,https://www.olympics.com/en/milano-cortina-202...
12,Tesero Cross-Country Skiing Stadium,TCC,"[""discipline-cross-country-skiing"",""discipline...","[""CCS"",""NCB""]",NaN,https://www.olympics.com/en/milano-cortina-202...
13,Tofane Alpine Skiing Centre,CAL,discipline-alpine-skiing,ALP,https://www.olympics.com/en/sports/alpine-skiing/,https://www.olympics.com/en/milano-cortina-202...
14,Verona Olympic Arena,VOA,NaN,NaN,NaN,https://www.olympics.com/en/milano-cortina-202...


In [138]:
df_venues.describe()

,venue,venue_code,disciplines,disciplines_code,url_sport,url_venue
count,15,15,13,13,7,15
unique,15,15,12,12,6,15
top,Anterselva Biathlon Arena,ABA,discipline-ice-hockey,IHO,https://www.olympics.com/en/sports/ice-hockey/,https://www.olympics.com/en/milano-cortina-202...
freq,1,1,2,2,2,1


Name is either individuals or country.

# Quality check

## Athletes

In [11]:
duplicated_rows = df_athletes[df_athletes.duplicated()]
duplicated_rows

,code,name,gender,country_code,flag_bearer,events


In [12]:
df_athletes.nunique()

code            2916
name            2916
gender             2
country_code      93
flag_bearer        4
events           344
dtype: int64

Complete dataset for athletes. Extra flag bearer info.

## Coaches

In [ ]:
duplicated_rows = df_coaches[df_coaches.duplicated()]
duplicated_rows

,code,name,gender,country_code,flag_bearer,events


In [30]:
df_coaches.nunique()

code            50
name            50
gender           2
country_code    18
nationality     15
function         2
events           8
dtype: int64

Incomplete dataset for coaches (only 50).

## Medallists

In [ ]:
duplicated_rows = df_medallists[df_medallists.duplicated()]
duplicated_rows

,date,medal_code,medal,code,name,gender,country_code,country,discipline,discipline_code,event_name


In [ ]:
df_medallists.nunique()

date                16
medal_code           3
medal                3
code               534
name               534
gender               2
country_code        30
country             30
discipline          16
discipline_code     16
event_name         104
dtype: int64

Looks like 31 medals are missing. Can fill dataset by hand.

## Medals

In [ ]:
duplicated_rows = df_medals[df_medals.duplicated()]
duplicated_rows

,country,country_code,gold,silver,bronze,total,rank,rank_total


In [ ]:
df_medals.nunique()

country         29
country_code    29
gold            10
silver          11
bronze          12
total           17
rank            26
rank_total      17
dtype: int64

In [ ]:
df_medals["total"].sum()

np.int64(348)

Only one medal per sport not per individual (740 vs. 348). Complete sub-dataset.

## Schedules

In [72]:
duplicated_rows = df_schedules[df_schedules.duplicated()]
duplicated_rows

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
7,2026-02-04T11:30:00+01:00,2026-02-04T13:30:00+01:00,2026-02-04,FINISHED,Alpine Skiing,ALP,Men's Downhill,0,Men's Downhill Official Training,M,INDV,Stelvio Ski Centre,SSC,Stelvio Ski Centre-Alpine Skiing Course,SAL,ALPMDH----------------TRNO000100--
8,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet A,CUA,CURXTEAM2-------------PREL000101--
9,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet B,CUB,CURXTEAM2-------------PREL000102--
10,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet C,CUC,CURXTEAM2-------------PREL000103--
11,2026-02-04T19:05:00+01:00,2026-02-04T21:00:00+01:00,2026-02-04,FINISHED,Curling,CUR,Mixed Doubles,0,Mixed Doubles Round Robin,X,TEAM,Cortina Curling Olympic Stadium,CCU,Cortina Curling Olympic Stadium- Sheet D,CUD,CURXTEAM2-------------PREL000104--
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1879,2026-02-22T11:09:00+01:00,2026-02-22T11:34:00+01:00,2026-02-22,FINISHED,Freestyle Skiing,FRS,Women's Freeski Halfpipe,0,Women's Freeski Halfpipe Final,W,INDV,Livigno Snow Park,LSP,Livigno Snow Park-Halfpipe,LHP,FRSWHP----------------FNL-000200--
1880,2026-02-22T11:38:00+01:00,2026-02-22T12:03:00+01:00,2026-02-22,FINISHED,Freestyle Skiing,FRS,Women's Freeski Halfpipe,1,Women's Freeski Halfpipe Final,W,INDV,Livigno Snow Park,LSP,Livigno Snow Park-Halfpipe,LHP,FRSWHP----------------FNL-000300--
1881,2026-02-22T12:15:00+01:00,2026-02-22T13:00:00+01:00,2026-02-22,FINISHED,Bobsleigh,BOB,4-man,1,4-man Competition,M,TEAM,Cortina Sliding Centre,CSC,Cortina Sliding Centre,CSC,BOBMTEAM4-------------FNL-000400--
1882,2026-02-22T14:10:00+01:00,2026-02-22T16:40:00+01:00,2026-02-22,FINISHED,Ice Hockey,IHO,Men,1,Men's Play-off Round,M,TEAM,Milano Santagiulia Ice Hockey Arena,MH1,Milano Santagiulia Ice Hockey Arena,MH1,IHOMTEAM6-------------FNL-000100--


In [76]:
df_schedules_unique = df_schedules.drop_duplicates()
df_schedules_unique.shape

(628, 16)

In [78]:
df_schedules_unique.nunique()

start_date         485
end_date           503
day                 19
status               3
discipline          17
discipline_code     17
event              112
event_medal          3
phase              245
gender               4
event_type           5
venue               15
venue_code          15
location            25
location_code       25
id                 628
dtype: int64

In [80]:
df_schedules_unique.groupby("status").size()

status
CANCELLED      3
FINISHED     624
SCHEDULED      1
dtype: int64

In [97]:
phase_cancelled = df_schedules_unique[df_schedules_unique["status"] == "CANCELLED"]["phase"]
phase_cancelled

18      Women's Downhill Official Training
1246                      Men's Super Team
1353           Women's Team Pursuit Finals
Name: phase, dtype: str

In [99]:
df_schedules_unique[df_schedules_unique["phase"].isin(phase_cancelled)]

,start_date,end_date,day,status,discipline,discipline_code,event,event_medal,phase,gender,event_type,venue,venue_code,location,location_code,id
18,2026-02-05T11:30:00+01:00,NaN,2026-02-05,CANCELLED,Alpine Skiing,ALP,Women's Downhill,0,Women's Downhill Official Training,W,INDV,Tofane Alpine Skiing Centre,CAL,Tofane Alpine Skiing Centre,CAL,ALPWDH----------------TRNO000100--
76,2026-02-06T11:30:00+01:00,2026-02-06T13:30:00+01:00,2026-02-06,FINISHED,Alpine Skiing,ALP,Women's Downhill,0,Women's Downhill Official Training,W,INDV,Tofane Alpine Skiing Centre,CAL,Tofane Alpine Skiing Centre,CAL,ALPWDH----------------TRNO000200--
132,2026-02-07T11:30:00+01:00,2026-02-07T13:30:00+01:00,2026-02-07,FINISHED,Alpine Skiing,ALP,Women's Downhill,0,Women's Downhill Official Training,W,INDV,Tofane Alpine Skiing Centre,CAL,Tofane Alpine Skiing Centre,CAL,ALPWDH----------------TRNO000300--
1237,2026-02-16T18:00:00+01:00,2026-02-16T18:35:00+01:00,2026-02-16,FINISHED,Ski Jumping,SJP,Men's Super Team,0,Men's Super Team,M,DGRP,Predazzo Ski Jumping Stadium,PSJ,Predazzo Ski Jumping-Large Hill,PLH,SJPMLHTEAM2-----------FNL-TR01SJ--
1239,2026-02-16T19:00:00+01:00,2026-02-16T19:35:00+01:00,2026-02-16,FINISHED,Ski Jumping,SJP,Men's Super Team,0,Men's Super Team,M,DGRP,Predazzo Ski Jumping Stadium,PSJ,Predazzo Ski Jumping-Large Hill,PLH,SJPMLHTEAM2-----------FNL-0001SJ--
1244,2026-02-16T19:50:00+01:00,2026-02-16T20:16:00+01:00,2026-02-16,FINISHED,Ski Jumping,SJP,Men's Super Team,1,Men's Super Team,M,DGRP,Predazzo Ski Jumping Stadium,PSJ,Predazzo Ski Jumping-Large Hill,PLH,SJPMLHTEAM2-----------FNL-0002SJ--
1246,2026-02-16T20:31:00+01:00,NaN,2026-02-16,CANCELLED,Ski Jumping,SJP,Men's Super Team,0,Men's Super Team,M,DGRP,Predazzo Ski Jumping Stadium,PSJ,Predazzo Ski Jumping-Large Hill,PLH,SJPMLHTEAM2-----------FNL-0003SJ--
1353,2026-02-17T15:43:00+01:00,2026-02-17T15:49:00+01:00,2026-02-17,CANCELLED,Speed Skating,SSK,Women's Team Pursuit,0,Women's Team Pursuit Finals,W,TEAM,Milano Speed Skating Stadium,MSS,Milano Speed Skating Stadium,MSS,SSKWTEAMPU------------FNL-D00100--
1354,2026-02-17T15:49:00+01:00,2026-02-17T15:55:00+01:00,2026-02-17,FINISHED,Speed Skating,SSK,Women's Team Pursuit,0,Women's Team Pursuit Finals,W,TEAM,Milano Speed Skating Stadium,MSS,Milano Speed Skating Stadium,MSS,SSKWTEAMPU------------FNL-C00100--
1358,2026-02-17T16:41:00+01:00,2026-02-17T16:47:00+01:00,2026-02-17,FINISHED,Speed Skating,SSK,Women's Team Pursuit,3,Women's Team Pursuit Finals,W,TEAM,Milano Speed Skating Stadium,MSS,Milano Speed Skating Stadium,MSS,SSKWTEAMPU------------FNL-B00100--


Watch out for duplicates! Scheduled is the ceremony (see tail) which is also the reason for 17 "disciplines" even though only 16 took place. Drop cancelled events + ceremony (depending on analysis). Cancelled events took place afterwards.

## Teams

In [109]:
duplicated_rows = df_teams[df_teams.duplicated()]
duplicated_rows

,code,name,team_type,gender,country_code,events


In [110]:
df_teams.nunique()

code            550
name            205
team_type         5
gender            4
country_code     43
events           39
dtype: int64

In [113]:
df_teams.groupby("team_type").size()

team_type
CPLP       28
CPLW       59
CUSTOM     78
ORG       329
ORGN       56
dtype: int64

In [120]:
countries_athletes = set(df_athletes["country_code"])
countries_teams = set(df_teams["country_code"])

missing_countries = countries_athletes - countries_teams


missing_countries

{'AIN',
 'ALB',
 'AND',
 'AZE',
 'BEN',
 'BIH',
 'BOL',
 'CHI',
 'COL',
 'CYP',
 'ECU',
 'ERI',
 'GBS',
 'HAI',
 'HKG',
 'IND',
 'IRI',
 'IRL',
 'ISL',
 'KEN',
 'KGZ',
 'KOS',
 'KSA',
 'LBN',
 'LUX',
 'MAD',
 'MAR',
 'MAS',
 'MDA',
 'MEX',
 'MGL',
 'MKD',
 'MLT',
 'MNE',
 'MON',
 'NGR',
 'NZL',
 'PAK',
 'PHI',
 'POR',
 'PUR',
 'RSA',
 'SGP',
 'SMR',
 'SRB',
 'THA',
 'UAE',
 'URU',
 'UZB',
 'VEN'}

In [129]:
len(set(df_teams["events"].apply(lambda x: x[0]["discipline"])))

16

CPLP: CouPle Link Pair

CPLW: CouPle Link

CUSTOM: Individuals

ORG: Official Team

ORGN: Official Team (when multiple teams)

Requires further processing to complete gender data ("-" to "X"/"M"/"F") through "event" column.

Looks like team sub-dataset is incomplete. Missing countries representation. Use sub-dataset with care.

## Venues

In [139]:
duplicated_rows = df_venues[df_venues.duplicated()]
duplicated_rows

,venue,venue_code,disciplines,disciplines_code,url_sport,url_venue


In [140]:
df_venues.nunique()

venue               15
venue_code          15
disciplines         12
disciplines_code    12
url_sport            6
url_venue           15
dtype: int64

Need to manually complete sub-dataset. Should be quick as it is small.

# Ideas of exploration

Local map with events clickable to enter new page dedicated to sports which took place there

Look into past winter olympics and see dynasties/new comers in Milan 2026

Look into visitors/economical/environmentals impacts

Do animation where we see dots arriving to the games (1 dot per athlete): start animation

Classic medal race

Flag bearers (legendary figures)

Are coaches old legends? And do they work for their country OR the best country in the domain OR rich country.

Men vs Women in discipline per country

Team sports per country

Most exploited areas (see schedule)

Which sports have most medals (where to specialize if medal race is the goal)

Use sports/venue URL to redirect to Olympics official website

Don't forget to cite dataset!